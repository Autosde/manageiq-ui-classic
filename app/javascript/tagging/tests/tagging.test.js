import React from 'react';
import Tagging from '../components/tagging';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

const tags = { animal: ['duck', 'pig'], food: ['steak', 'salad'] };
const setTags = [{ tagCategory: 'animal', tagValue: 'pig' }, { tagCategory: 'food', tagValue: 'steak' }];
const selectedTagCategory = 'animal';
const selectedTagValue = 'pig';
function onChange(x) {
  return x;
}
function onDelete(x) {
  return x;
}


describe('Tagging component without redux mapping', () => {
  it('match snapshot', () => {
    const component = renderer.create(<Tagging
      tags={tags}
      setTags={setTags}
      onTagValueChange={onChange}
      onTagCategoryChange={onChange}
      onTagDeleteClick={onDelete}
      selectedTagCategory={selectedTagCategory}
      selectedTagValue={selectedTagValue}
    />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call methods', () => {
    const onTagCategoryChange = jest.fn();
    const onTagValueChange = jest.fn();
    const onTagDeleteClick = jest.fn();
    const wrapper  = shallow(<Tagging
      tags={tags}
      setTags={setTags}
      onTagValueChange={onTagValueChange}
      onTagCategoryChange={onTagCategoryChange}
      onTagDeleteClick={onTagDeleteClick}
      selectedTagCategory={selectedTagCategory}
      selectedTagValue={selectedTagValue}
    />);
    wrapper.instance().onTagCategoryChange('xaxa');
    expect(onTagCategoryChange.mock.calls.length).toEqual(1);
    wrapper.instance().onTagValueChange('wawa');
    expect(onTagCategoryChange.mock.calls.length).toEqual(1);
    wrapper.instance().onTagDeleteClick('wowo');
    expect(onTagCategoryChange.mock.calls.length).toEqual(1);
  })
})
