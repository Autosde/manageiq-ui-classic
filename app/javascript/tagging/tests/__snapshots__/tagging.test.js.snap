// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tagging component without redux mapping match snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-lg-6"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-12"
        >
          <h1>
            Add/modify tag
          </h1>
        </div>
        <div
          className="col-lg-6"
        >
          <h2>
            Category:
          </h2>
        </div>
        <div
          className="col-lg-6"
        >
          <div
            className="Select has-value is-clearable is-searchable Select--single"
            style={undefined}
          >
            <input
              disabled={false}
              name="form-field-name"
              type="hidden"
              value="animal"
            />
            <div
              className="Select-control"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={undefined}
            >
              <span
                className="Select-multi-value-wrapper"
                id="react-select-2--value"
              >
                <div
                  className="Select-value"
                  style={undefined}
                  title={undefined}
                >
                  <span
                    aria-selected="true"
                    className="Select-value-label"
                    id="react-select-2--value-item"
                    role="option"
                  >
                    animal
                  </span>
                </div>
                <div
                  className="Select-input"
                  style={
                    Object {
                      "display": "inline-block",
                    }
                  }
                >
                  <input
                    aria-activedescendant="react-select-2--value"
                    aria-describedby={undefined}
                    aria-expanded="false"
                    aria-haspopup="false"
                    aria-label={undefined}
                    aria-labelledby={undefined}
                    aria-owns=""
                    className={undefined}
                    id={undefined}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    role="combobox"
                    style={
                      Object {
                        "boxSizing": "content-box",
                        "width": "5px",
                      }
                    }
                    tabIndex={undefined}
                    value=""
                  />
                  <div
                    style={
                      Object {
                        "height": 0,
                        "left": 0,
                        "overflow": "scroll",
                        "position": "absolute",
                        "top": 0,
                        "visibility": "hidden",
                        "whiteSpace": "pre",
                      }
                    }
                  >
                    
                  </div>
                </div>
              </span>
              <span
                aria-label="Clear value"
                className="Select-clear-zone"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                title="Clear value"
              >
                <span
                  className="Select-clear"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&times;",
                    }
                  }
                />
              </span>
              <span
                className="Select-arrow-zone"
                onMouseDown={[Function]}
              >
                <span
                  className="Select-arrow"
                  onMouseDown={[Function]}
                />
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-lg-6"
        >
          <h2>
            Assigned Value:
          </h2>
        </div>
        <div
          className="col-lg-6"
        >
          <div
            className="Select has-value is-clearable is-searchable Select--single"
            style={undefined}
          >
            <input
              disabled={false}
              name="form-field-name"
              type="hidden"
              value="pig"
            />
            <div
              className="Select-control"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={undefined}
            >
              <span
                className="Select-multi-value-wrapper"
                id="react-select-3--value"
              >
                <div
                  className="Select-value"
                  style={undefined}
                  title={undefined}
                >
                  <span
                    aria-selected="true"
                    className="Select-value-label"
                    id="react-select-3--value-item"
                    role="option"
                  >
                    pig
                  </span>
                </div>
                <div
                  className="Select-input"
                  style={
                    Object {
                      "display": "inline-block",
                    }
                  }
                >
                  <input
                    aria-activedescendant="react-select-3--value"
                    aria-describedby={undefined}
                    aria-expanded="false"
                    aria-haspopup="false"
                    aria-label={undefined}
                    aria-labelledby={undefined}
                    aria-owns=""
                    className={undefined}
                    id={undefined}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    role="combobox"
                    style={
                      Object {
                        "boxSizing": "content-box",
                        "width": "5px",
                      }
                    }
                    tabIndex={undefined}
                    value=""
                  />
                  <div
                    style={
                      Object {
                        "height": 0,
                        "left": 0,
                        "overflow": "scroll",
                        "position": "absolute",
                        "top": 0,
                        "visibility": "hidden",
                        "whiteSpace": "pre",
                      }
                    }
                  >
                    
                  </div>
                </div>
              </span>
              <span
                aria-label="Clear value"
                className="Select-clear-zone"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                title="Clear value"
              >
                <span
                  className="Select-clear"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&times;",
                    }
                  }
                />
              </span>
              <span
                className="Select-arrow-zone"
                onMouseDown={[Function]}
              >
                <span
                  className="Select-arrow"
                  onMouseDown={[Function]}
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-lg-6"
    >
      <ul
        className="list-inline"
      >
        <li>
          <span
            className="label label-info"
          >
            animal
            : 
            pig
            <a
              href="#"
              onClick={[Function]}
            >
              <span
                aria-hidden="true"
                className="pficon pficon-close"
              />
              <span
                className="sr-only"
              >
                Remove
              </span>
            </a>
          </span>
        </li>
        <li>
          <span
            className="label label-info"
          >
            food
            : 
            steak
            <a
              href="#"
              onClick={[Function]}
            >
              <span
                aria-hidden="true"
                className="pficon pficon-close"
              />
              <span
                className="sr-only"
              >
                Remove
              </span>
            </a>
          </span>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
